mqtt/mqtt-bridge/src/settings.rs

```diff
@@ -326,181 +312,53 @@ impl RingBufferSettings {
-        let storage_settings = settings.storage();
-        // Should exist from default.json.
-        assert_matches!(storage_settings, StorageSettings::RingBuffer(_));
-        if let StorageSettings::RingBuffer(rb) = storage_settings {
-            assert_eq!(rb.max_file_size(), NonZeroU64::new(33_554_432).unwrap());
-            assert_eq!(*rb.directory(), PathBuf::from("/tmp/mqttd/"));
-            assert_eq!(*rb.flush_options(), FlushOptions::AfterEachWrite);
-        }
-    }
-
-    #[test]
-    #[serial(env_settings)]
-    fn from_file_reads_nested_bridge_settings() {
-        let settings = BridgeSettings::from_file("tests/config.json").unwrap();
-        let upstream = settings.upstream().unwrap();
-
-        assert_eq!(upstream.name(), "$upstream");
-        assert_eq!(upstream.address(), "edge1:8883");
-
-        match upstream.credentials() {
-            Credentials::Provider(provider) => {
-                assert_eq!(provider.iothub_hostname(), "iothub");
-                assert_eq!(provider.device_id(), "d1");
-                assert_eq!(provider.module_id(), "mymodule");
-                assert_eq!(provider.generation_id(), "321");
-                assert_eq!(provider.workload_uri(), "uri");
-            }
-            _ => panic!("Expected provider settings"),
-        };
-    }
-
-    #[test]
-    #[serial(env_settings)]
-    fn from_file_reads_remotes_settings() {
-        let settings = BridgeSettings::from_file("tests/config.json").unwrap();
-        let len = settings.remotes().len();
-
-        assert_eq!(len, 1);
-        let remote = settings.remotes().first().unwrap();
-        assert_eq!(remote.name(), "r1");
-        assert_eq!(remote.address(), "remote:8883");
-        assert_eq!(remote.keep_alive().as_secs(), 60);
-        assert_eq!(remote.clean_session(), false);
-
-        match remote.credentials() {
-            Credentials::PlainText(auth_settings) => {
-                assert_eq!(auth_settings.username(), "mymodule");
-                assert_eq!(auth_settings.password(), "pass");
-                assert_eq!(auth_settings.client_id(), "client");
-            }
-            _ => panic!("Expected plaintext settings"),
-        };
-    }
-
-    #[test]
-    #[serial(env_settings)]
-    fn from_file_reads_storage_settings_without_explicit_storage() {
-        let settings = BridgeSettings::from_file("tests/config.json").unwrap();
-        let storage_settings = settings.storage();
-        // Should exist from default.json.
-        assert_matches!(storage_settings, StorageSettings::RingBuffer(_));
-        if let StorageSettings::RingBuffer(rb) = storage_settings {
-            assert_eq!(rb.max_file_size(), NonZeroU64::new(33_554_432).unwrap());
-            assert_eq!(*rb.directory(), PathBuf::from("/tmp/mqttd/"));
-            assert_eq!(*rb.flush_options(), FlushOptions::AfterEachWrite);
+fn deserialize_nonzerou64<'de, D>(deserializer: D) -> Result<NonZeroU64, D::Error>
```

Comment:
serde doesn't support NonZero types deserializing?

Reply:
The point of these is to allow for multiple formats of the value (string/int) and then take the value and make it NonZeroX. I found this is nice for testing with env::set where things are strings and also gave some flexibility into how one wants to organize their values.